ocean/transects/python/create_transect_masks.py

```diff
@@ -0,0 +1,162 @@
+
+    ds_mask = compute_mpas_transect_masks(dsMesh=ds_mesh, fcMask=fc_transect,
+                                          earthRadius=earth_radius,
+                                          maskTypes=('cell', 'edge',),
+                                          logger=logger,
+                                          pool=pool, addEdgeSign=True)
+
+    feature = fc_transect.features[0]
+    geom_type = feature['geometry']['type']
+    if geom_type == 'LineString':
+        coordinates = [feature['geometry']['coordinates']]
+    elif geom_type == 'MultiLineString':
+        coordinates = feature['geometry']['coordinates']
+    else:
+        raise ValueError(
+            f'Unexpected geometry type for the transect {geom_type}')
+
+    lon = []
+    lat = []
+    for coords in coordinates:
+        lon_local, lat_local = zip(*coords)
+        lon.extend(lon_local)
+        lat.extend(lat_local)
+    lon = xr.DataArray(data=lon, dims='nTransectPoints')
+    lat = xr.DataArray(data=lat, dims='nTransectPoints')
+
+    layer_thickness = ds_mesh.layerThickness
+    bottom_depth = ds_mesh.bottomDepth
+    min_level_cell = ds_mesh.minLevelCell
+    max_level_cell = ds_mesh.maxLevelCell
+
+    ds_transect = compute_transect(lon, lat, ds_mesh, layer_thickness,
+                                   bottom_depth, min_level_cell,
+                                   max_level_cell, spherical=True)
+
+    ds = ds_mask
+    for var in ds_transect.data_vars:
+        ds[var] = ds_transect[var]
+
+    add_distance_field(ds, logger)
+
+    write_netcdf(ds, out_filename)
+
+
+def add_distance_field(ds, logger):
+    """
+    Add fields on edges and cells that define the (mean) distance along the
+    transect for each cell or edge in the transect
+    """
+
+    dist_cell = np.zeros(ds.sizes['nCells'])
+    count_cell = np.zeros(ds.sizes['nCells'], dtype=int)
+    dist_edge = np.zeros(ds.sizes['nEdges'])
+    count_edge = np.zeros(ds.sizes['nEdges'], dtype=int)
+
+    logger.info('Adding transect distance fields on cells and edges...')
+
+    for segment in range(ds.sizes['nSegments']):
+        icell = ds.horizCellIndices.isel(nSegments=segment).values
+        iedge = ds.horizEdgeIndices.isel(nSegments=segment).values
+        # the distance for the midpoint of the segment is the mean
+        # of the distances of the end points
+        dist = 0.5 * (ds.dNode.isel(nHorizNodes=segment) +
+                      ds.dNode.isel(nHorizNodes=segment + 1))
+        dist_cell[icell] += dist
+        count_cell[icell] += 1
+        dist_edge[iedge] += dist
+        count_edge[iedge] += 1
+
+    mask = count_cell > 0
+    dist_cell[mask] /= count_cell[mask]
+    dist_cell[np.logical_not(mask)] = np.nan
+
+    mask = count_edge > 0
+    dist_edge[mask] /= count_edge[mask]
+    dist_edge[np.logical_not(mask)] = np.nan
+
+    ds['transectDistanceCell'] = ('nCells', dist_cell)
+    ds['transectDistanceEdge'] = ('nEdges', dist_edge)
+    logger.info('Done.')
```

Comment:
To me, this part is the important new addition.

The `transect` module is just here because the version in Polaris (which this is a copy of) is cleaner than the version in MPAS-Tools but we don't have a way to use it in QuickViz because Polaris isn't a simple conda package that can be installed from conda-forge.